azure-pipelines.yml

```diff
@@ -0,0 +1,47 @@
+trigger:
+  - master
+
+pool:
+  vmImage: 'ubuntu-latest'
+
+variables:
+  - group: containergroup
+  - name: containerImage
+    value: elm-lmp-api:$(Build.BuildId)
+  - name: repository
+    value: elm-lmp-api
+  - name: testOutputDir
+    value: test-output
+
+steps:
+- task: Docker@2
+  displayName: Build production image
+  inputs:
+    command: build
+    containerRegistry: $(azureContainerRegistry)
+    repository: $(repository)
+    arguments: -t $(containerImage)
+
+- bash: docker build -t $(containerImage)-development --target=development .
+  displayName: Build development image
+
+- bash: bin/test
+  displayName: Test
+  env:
+    CONTAINER_IMAGE: $(containerImage)
+    OUTPUT_DIR: $(testOutputDir)
```

Comment:
rather than a bash script running the docker command line tool this could be done cleaner with a docker compose file that mounts the local volume:     

>     mkdir test-output
>     chmod 777 test-output
>     docker-compose  -f docker-compose-test.yml build --no-cache test
>     docker-compose  -f docker-compose-test.yml up --force-recreate test

it could potentially be split into three tasks. a `script` to create the folder, a `DockerCompose@0` to build the test image and `DockerCompose@0` to run the test image, giving better visibility of which action failed if there are problems.

There's a trade off between having the same scripts locally as on the build server, and being able to see immediately what part of the build has failed without having to look into logs. One to have a team discussion about.

Reply:
I've changed back to using Docker Compose now that we worked out what was wrong with the task type in Azure Pipelines. The build and test scripts are now one liners that run docker-compose commands, with equivalent behaviour achieved through DockerCompose tasks in the pipeline.